plugins/periskop/src/components/PeriskopErrorsTable/PeriskopErrorsTable.tsx

```diff
@@ -0,0 +1,190 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import moment from 'moment';
+import { Entity } from '@backstage/catalog-model';
+import { useApi } from '@backstage/core-plugin-api';
+import { useEntity } from '@backstage/plugin-catalog-react';
+import {
+  Content,
+  ContentHeader,
+  Table,
+  TableColumn,
+  Progress,
+  StatusWarning,
+  StatusError,
+  StatusPending,
+  Select,
+  EmptyState,
+  Link,
+} from '@backstage/core-components';
+import Alert from '@material-ui/lab/Alert';
+import useAsync from 'react-use/lib/useAsync';
+import { periskopApiRef } from '../..';
+import { AggregatedError, NotFoundInLocation } from '../../types';
+
+/**
+ * Constant storing Periskop project name.
+ *
+ * @public
+ */
+export const PERISKOP_NAME_ANNOTATION = 'periskop.io/name';
+
+/**
+ * Returns true if Periskop annotation is present in the given entity.
+ *
+ * @public
+ */
+export const isPeriskopAvailable = (entity: Entity): boolean =>
+  Boolean(entity.metadata.annotations?.[PERISKOP_NAME_ANNOTATION]);
+
+const renderKey = (
+  error: AggregatedError,
+  linkTarget: string,
+): React.ReactNode => {
+  return (
+    <Link to={linkTarget} target="_blank" rel="noreferrer">
+      {error.aggregation_key}
+    </Link>
+  );
+};
+
+const renderSeverity = (severity: string): React.ReactNode => {
+  if (severity.toLocaleLowerCase('en-US') === 'warning') {
+    return <StatusWarning>{severity}</StatusWarning>;
+  } else if (severity.toLocaleLowerCase('en-US') === 'error') {
+    return <StatusError>{severity}</StatusError>;
+  }
+  return <StatusPending>{severity}</StatusPending>;
+};
+
+const renderLastOccurrence = (error: AggregatedError): React.ReactNode => {
+  return moment(new Date(error.latest_errors[0].timestamp * 1000)).fromNow();
+};
+
+function isNotFoundInLocation(
+  apiResult: AggregatedError[] | NotFoundInLocation | undefined,
+): apiResult is NotFoundInLocation {
+  return (apiResult as NotFoundInLocation)?.body !== undefined;
+}
+
+export const PeriskopErrorsTable = () => {
+  const { entity } = useEntity();
+  const entityPeriskopName: string =
+    (entity.metadata.annotations?.[PERISKOP_NAME_ANNOTATION] as
+      | string
+      | undefined) ?? entity.metadata.name;
+
+  const periskopApi = useApi(periskopApiRef);
+  const locations = periskopApi.getLocationNames();
+  const [locationOption, setLocationOption] = React.useState<string>(
+    locations[0],
+  );
+  const {
+    value: aggregatedErrors,
+    loading,
+    error,
+  } = useAsync(async (): Promise<AggregatedError[] | NotFoundInLocation> => {
+    return periskopApi.getErrors(locationOption, entityPeriskopName);
+  }, [locationOption]);
+
+  if (loading) {
+    return <Progress />;
+  } else if (error) {
+    return <Alert severity="error">{error.message}</Alert>;
```

Comment:
there's also a ResponseErrorPanel in core-components that can show full tracebacks and stuff upon expanding, if you like that kind of thing. It can even transfer error state from the backend all the way to the frontend, if the backend throws a `@backstage/errors` error